Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -301,57 +282,37 @@ public void Generate_InvalidSqaleFileSpecified_GeneratorError()
 
         #region Private methods
 
-        private AnalyzerPluginGenerator CreateTestSubjectWithFakeRemoteRepo(RemoteRepoBuilder remoteRepoBuilder)
+        private static ProcessedArgs CreateArgs(string packageId, string packageVersion, string language, string sqaleFilePath, bool acceptLicenses, string outputDirectory)
         {
-            TestLogger logger = new TestLogger();
-            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(remoteRepoBuilder.FakeRemoteRepo, GetLocalNuGetDownloadDir(), logger);
-            return new AnalyzerPluginGenerator(nuGetHandler, logger);
+            ProcessedArgs args = new ProcessedArgs(
+                packageId,
+                new SemanticVersion(packageVersion),
+                language,
+                sqaleFilePath,
+                acceptLicenses,
+                outputDirectory);
+            return args;
         }
 
-        /// <summary>
-        /// Creates a graph used for testing, with nodes labelled in breadth-first order.
-        /// 
-        /// Visually:
-        /// Root--------
-        /// |            \
-        /// Child1       Child2---------
-        /// |             |             \
-        /// Grandchild1_1 GrandChild2_1 GrandChild2_2
-        /// </summary>
-        /// <param name="nodesRequireLicense">
-        /// Nodes in the graph that should be packages with the field requireLicenseAccept set to true
-        /// </param>
-        private void SetupTestGraph(RemoteRepoBuilder remoteRepoBuilder, params Node[] nodesRequireLicense)
```

Comment:
The new tests are slightly simpler as the dependency walking is tested elsewhere (as part of the NuGetPackageHandler tests). The new tests don't all use the set of packages; they explicitly create the package layout they need for that particular test.